### PR TITLE
Fix tests.

### PR DESCRIPTION
**Pull request by dichenqiandbt**
## What is this PR?
This is a:
- [ ] documentation update
- [x] bug fix with no breaking changes
- [ ] new functionality
- [ ] a breaking change

All pull requests from community contributors should target the `main` branch (default).

## Description & motivation
Fix tests.
- Replace old env setup helper function(buggy) with DbtCoreTestBase. The idea is that I noticed lot's of tests will set dbt core envs and cleanup in tearDown, so I refactor it out of test class.
- test_compilation.py, adopts new DbtCoreTestBase to fix bug. It requires to install dbt-postgres, I added this dependency in dev requirements.
- test_state.py, adopts new DbtCoreTestBase to fix bug,
- test_views.py, adopts new DbtCoreTestBase to fix bug.
- test_cache.py, add get_latest_project_path patch to ensure tests are not related to local cache files.

## Checklist
- [ ] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [ ] Postgres
    - [ ] Redshift
    - [ ] Snowflake
    - [ ] Databricks
    - [ ] Spark
- [ ] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models
- [ ] I have added an entry to CHANGELOG.md

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,4 +1,6 @@
 black==22.6.0
+dbt-postgres==1.4.1
```
Comment (racheldaniel) [inline dev-requirements.txt:2]: I think we want to either let the developer install dbt and an adapter on their own, or let this install latest and have developers override the version. Server v0.2.0 will only be used with 1.5+, so at the very least we'd want to install `1.5.0b1` here and use the `1.4.1` installation on the `0.1.latest` branch.

Alternatively, we could have a wrapper for the `pytest` command for local dev that accepts a dbt version and installs it and the adapters, or defaults to a dbt version expected to work with that server version

**Review by racheldaniel — CHANGES_REQUESTED** [review]
Left a comment about adapter installation-- once we figure that out this is good to go!

File: dev-requirements.txt (hunk 1 above)
Comment (dichenqiandbt) [reply dev-requirements.txt]: Offline chatted with Rachel, we tended to
- Keep core and adaptors as external dependency, hence I removed adaptor dependency.
- However some e2e tests require such dependency, added skip rule to skip those tests.

**Review by racheldaniel — APPROVED** [review]
LGTM!

**Pull request merged**